### DISCUSSION TITLE
!#zh: 欢迎来到 鲁班H5 讨论区

### DISCUSSION THREAD

**Discussion by ly525** [ly525/luban-h5#216] [General]
## 👋 Welcome!
大家好，欢迎大家来到鲁班H5 社区/论坛/讨论区，我们希望让每一次交流被沉淀！
因为微信群等聊天工具有一些痛点：
 - 聊天内容有些分散，并不一定是技术 + 产品相关内容，有很多无意义内容
 - 新入群的同学无法快速融入，不了解上下文
 - 聊天内容，无法作为社区好的沉淀
 - 等等
 
因此，我们申请了 [Discussions](https://github.com/ly525/luban-h5/discussions) 作为鲁班H5 的社区/论坛/讨论区。我们希望你:

 * 问一些你想知道的问题
 * 分享你的想法
 * 与社区里其它同学接触、交流
 * 欢迎别人并且开放思想。记住这是一个我们大家共建的社区
 * 将心比心，你认真提问，大家认真解答
 * 务必阅读：[如何提问，并获得高回复率——提问的智慧](https://zhuanlan.zhihu.com/p/19779979)

## 简介
大家好，我是鲁班H5的作者+维护者，为了能够让社区更加高效、健康的运转下去，我们申请开通了希望能够从社区中吸引开发者来共同维护，进行社区共建。

鲁班H5 作为一个低代码开发解决方案，还处在初级的发展阶段，是一个发展中的方案，在不久的未来我们会有更多新的功能、特性推向开发者/用户，甚至可能也有颠覆性的东西出现，这很可能来自你的一个想法/建议/评论，让 鲁班H5 成为一个更棒的方案。目前 鲁班H5 已经翻过了一座座发展的山峰，但他正在向更高山峰迈进。

长风破浪会有时，直挂云帆济沧海。

## 交流边界
 * 低代码、可视化搭建相关产品、技术方向
 * 低代码、可视化搭建相关技术
 * 鲁班H5 相关技术、产品
 * 不划水、发无意义内容
 
 ## 社区常见问题答疑
 1. 鲁班是个人项目还是公司项目？
答：个人项目，非全职，业余时间做的项目

2. 我们希望加一些数据统计方面的需求（PV、UV、点击热点、访问分布）
答：PV 统计等方案，可以借助 Matomo 来做（鲁班后期会结合 Matomo 做一些无痕打点的尝试）

3. 目前无法发布？
答：发布涉及到权限控制 + 用户模块，这个我想等到主体功能完毕之后，再添加用户模块

4. 后续的开发计划是什么？
答：[Projects](https://github.com/ly525/luban-h5/projects)
   - 完善组件生态，支持快速开发组件
   - 动作模块，支持页面跳转等行为
   - 支持数据拉取，打通和后端通信
   - 支持模板引擎，实现页面的千人千面。
   - JSON 配置生成 B端管理后台
   - 拖拽生成小程序的项目
   
    目前上面的技术方向，都有明确的实现思路了，所以打算先按照这个做下去

5. 目前有哪些困难呢？
答：时间不如可能是最大的困难了，剩下的是一些中高级的后端需求不一定能 hold 住

6. 现在能商用吗？或者有商用版本吗？
答：我个人认为 鲁班目前不足以达到商业化的标准。

7. 鲁班的team 能介绍下吗？
答：鲁班现在的核心开发只有我一个，非常感谢各位 [contributor](https://github.com/ly525/luban-h5/graphs/contributors) 的贡献！🙇‍♀️🙇

8. 一些期待：
答：因为鲁班现在还不完善，我个人经验和能力也有限。
您在实际使用过程中如果发现问题或者建议，在群里直接提问当然也是OK的，更欢迎在 GitHub 提 [issue](https://github.com/ly525/blog/issues)，以及在 [社区](https://github.com/ly525/luban-h5/discussions)交流

还请多多理解，以及多都给我反馈！

**Comment by ly525** [comment]
issue负责解决bug和feature，其他的讨论都在Discussion中进行。相关 bug 反馈 和 新 feature，请前往 [issue](https://github.com/ly525/luban-h5/issues)

**Comment by kinka** [comment]
hi~
关于组件生态有些疑问，不知道是还没实现，还是我操作不对。

1. 用创建组件的辅助工具发布了个demo组件，后续流程跑不起来，比如说拖放到编辑区没有展示维护，也无法进行属性编辑
2. 通过npm列表引入的组件，在最后保存的时候是如何与页面构建到一起的？

**Reply by ly525** [reply]
可以提交个代码仓库，发一下GitHub 或者 gitee 地址，我看下

**Comment by leochen-g** [comment]
有考虑维护创建一个模版社区么，可以把用鲁班做出来的模版分享出来供大家使用

**Reply by ly525** [reply]
考虑的，你意思是类似于 易企秀 之类的模版列表（包含社区贡献）对吗？
这里面有一个需要特殊注意的是：字体、图片版权，到时候加一个声明吧

**Reply by leochen-g** [reply]
嗯 是的，因为每个模版其实都是一个json数据组合的，可以做一个类似echarts社区实例的那种社区，然后客户端功能里加一个模版导入功能，或者是后台里直接导入也可以

**Reply by leochen-g** [reply]
只是这种模版分享的形式，可能还是需要仔细思考一下如何去做

**Comment by dxc0522** [comment]
开发插件这个你们试过么？
luban-h5/vue-cli-plugin-lbhc  这个安装下来的依赖有冲突手动解决后打包传到npm然后在鲁班h5中进行引入后拖拽就会报错，说没有classList是null
我引入的方式是yarn add下载然后在配置NPM组件列表配置信息中加入，但是无法正常使用
我创建的包名叫  d-luban-color-button
